crates/connectors/ndc-postgres/src/configuration.rs

```diff
@@ -21,15 +23,56 @@ pub const CURRENT_VERSION: u32 = 1;
 /// since it consists of a sub-selection of components from the full Configuration, the fields are
 /// borrowed rather than owned.
 #[derive(Debug)]
-pub struct RuntimeConfiguration<'a> {
-    pub metadata: &'a metadata::Metadata,
+pub struct RuntimeConfiguration {
+    pub metadata: metadata::Metadata,
 }
 
 impl<'a> version1::Configuration {
     /// Apply the common interpretations on the Configuration API type into an RuntimeConfiguration.
-    pub fn as_runtime_configuration(self: &'a Configuration) -> RuntimeConfiguration<'a> {
+    pub fn as_runtime_configuration(self: &'a Configuration) -> RuntimeConfiguration {
         RuntimeConfiguration {
-            metadata: &self.config.metadata,
+            metadata: metadata_to_current(&self.config.metadata),
```

Comment:
All I mean is, currently we take the whole `Configuration` and derive `RuntimeConfiguration` from it on each request, surely we could just do this once?

It's not really relevant to this PR, mind, just something I noticed because you made a change near the code that does this.